tests/posix/getopt/testcase.yaml

```diff
@@ -1,9 +1,18 @@
+common:
+  arch_exclude: posix
+  tags: posix getopt
 tests:
-  shell.getopt:
-    tags: getopt
-    platform_exclude: native_posix native_posix_64
+  portability.posix.getopt:
     integration_platforms:
       - qemu_x86
     min_flash: 64
     min_ram: 32
-    filter: (CONFIG_GETOPT and not CONFIG_NEWLIB_LIBC)
+  portability.posix.getopt.newlib:
```

Comment:
Do these need min_flash/min_ram tests too?

Reply:
Probably not, to be honest. I could do another pass addressing ram sizes at another point.